src/NJsonSchema/Generation/JsonSchemaGenerator.cs

```diff
@@ -915,7 +915,11 @@ private async Task LoadPropertyOrFieldAsync(Newtonsoft.Json.Serialization.JsonPr
 
                 var propertyName = GetPropertyName(property, propertyInfo);
                 if (parentSchema.Properties.ContainsKey(propertyName))
+                {
+                    if (Settings.GenerateAbstractProperties && Settings.FlattenInheritanceHierarchy) return; // accept first instance of the duplicate
```

Comment:
I think we should use GetActualFlattenInheritanceHierarchy(parentType) instead of FlattenInheritanceHierarchy

Reply:
That makes sense. Have updated.